Set facet icons from CSS background-image property (DLME #1347)

marlo-longley opened the pull request:
Fixes [#1347](https://github.com/sul-dlss/dlme/issues/1347) from the DLME repo. The solution proposed was to make the change in this upstream gem.

I've used the Font Awesome plus/minus icons and added attribution in the README.

The visual change looks like this:
![Screen Shot 2022-02-08 at 14 56 45](https://user-images.githubusercontent.com/1328900/153094284-62206108-3b99-4276-b7bd-7dd79ef430a9.png)
![Screen Shot 2022-02-08 at 14 56 53](https://user-images.githubusercontent.com/1328900/153094292-355f0a29-d22f-4a42-8e91-b4bce56dc374.png)

jcoyne commented on app/assets/stylesheets/blacklight/hierarchy/hierarchy.scss:
```diff
@@ -34,8 +34,21 @@ $text-muted: #777 !default;
+
   .twiddle>.toggle-handle .closed {
-    display: inline;
+    background-image: url('data:image/svg+xml, <svg aria-hidden="true" focusable="false" data-prefix="far" data-icon="plus-square" class="svg-inline--fa fa-plus-square fa-w-14" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M352 240v32c0 6.6-5.4 12-12 12h-88v88c0 6.6-5.4 12-12 12h-32c-6.6 0-12-5.4-12-12v-88h-88c-6.6 0-12-5.4-12-12v-32c0-6.6 5.4-12 ...
```
Would it be smart to put this in a sass variable?

cbeer replied:
👍 ; that'd make it easier for downstream consumers to configure too.

cbeer replied:
I think it'd also be good and appropriate to retain the fontawesome attribution in the SVG content itself. The license on https://github.com/FortAwesome/Font-Awesome#license suggests that's sufficient attribution 🤷‍♂️

jcoyne commented on blacklight-hierarchy.gemspec:
```diff
@@ -28,4 +28,5 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'engine_cart', '~> 2.3'
   s.add_development_dependency 'capybara'
   s.add_development_dependency 'coveralls'
+  s.add_development_dependency 'net-smtp'
```
Huh?

marlo-longley replied:
I added this to run tests locally, my specific versions of rails and ruby made it necessary. Good call; it should not be included in this PR.

jcoyne commented on lib/blacklight/hierarchy/engine.rb:
```diff
@@ -4,8 +4,6 @@
 module Blacklight
   module Hierarchy
     class Engine < Rails::Engine
-      config.closed_icon = '⊞'
```
This would be a breaking change requiring a major version bump.

thatbudakguy replied:
What functionality is breaking?

cbeer replied:
I think that's probably fine -- especially given the very limited number of consumers (none of which customize this icon?).

jcoyne replied:
Major version increments are free though

marlo-longley commented:
Thanks for taking the time to look over this @jcoyne. I implemented the SASS variables, thanks for that idea. I also removed the `net-smtp` gem addition, but then the CI tests failed here. Happy to open a separate PR for editing the Gemfile.

cbeer commented:
I think #60 ought to fix the build for you, maybe?

jcoyne commented on app/assets/stylesheets/blacklight/hierarchy/hierarchy.scss:
```diff
@@ -1,3 +1,8 @@
+// plus sign
+$closed-icon: '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc. --><path d="M200 344V280H136C122.7 280 112 269.3 112 256C112 242.7 122.7 232 136 232H200V168C200 154.7 210.7 144 2 ...
```
please add `!default` to these, so we can override them.  How about prefixing with `b-h-` to avoid collisions with other variables?
Can you move the `url()` part here to as is done in Bootstrap (https://github.com/twbs/bootstrap/blob/main/scss/_variables.scss#L874)?
